logstash-core/lib/logstash/pipeline.rb

```diff
@@ -418,14 +418,14 @@ def worker_loop(batch_size, batch_delay)
       signal = @signal_queue.poll || NO_SIGNAL
       shutdown_requested |= signal.shutdown? # latch on shutdown signal
 
-      batch = @filter_queue_client.read_batch # metrics are started in read_batch
-      batch_size = batch.size
+      batch = @filter_queue_client.read_batch.to_java # metrics are started in read_batch
```

Comment:
@danhermann `read_batch.to_java` is strange? Why do we need this, I don't think we do right? It should unwrap automatically

Reply:
The `pipeline_spec` tests fail if `to_java` is not explicitly specified there.

Reply:
@danhermann ok :) let's just leave it then